livre/core/pipeline/Filter.h

```diff
@@ -0,0 +1,73 @@
+/* Copyright (c) 2011-2016, EPFL/Blue Brain Project
+ *                     Ahmet Bilgili <ahmet.bilgili@epfl.ch>
+ *
+ * This file is part of Livre <https://github.com/BlueBrain/Livre>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef _Filter_h_
+#define _Filter_h_
+
+#include <livre/core/types.h>
+#include <livre/core/pipeline/PortInfo.h>
+
+namespace livre
+{
+
+/**
+ * Filters are similar to functions (immutable). Their inputs and
+ * outputs are provided through ports. These ports provide
+ * thread safe information retrieval, setting and querying.
+ *
+ * One input port may have many incoming connections and one output
+ * port can be connected to many other input connections. The ports
+ * have their unique names and these names are used to query,
```

Comment:
Since this is an abstract class, the statement "ports have ~~their~~ unique names" makes me think that something enforces this is condition. If I write a derived class, is it possible for my to use the same name twice by mistake? what happens then?